src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -145,7 +145,7 @@ private void BroadcasterManager_TransactionStateChanged(object sender, Transacti
         {
             if (transactionEntry.State == State.Propagated)
             {
-                this.ProcessTransaction(transactionEntry.Transaction);
+                this.ProcessTransaction(transactionEntry.Transaction, null, null, true);
```

Comment:
This condition shouldn't matter really as the same transaction can be processed as many times as we want but ProcessTransaction is idempotent so it won't change the state of the wallet.
So, with this in mind we can:
1. remove this condition and just call ProcessTransaction
2. review what happens in AddOrUpdate, because we can always ProcessTransaction, irrespective of state.
3. rethink whether keeping a list of broadcasts and the AddOrUpdate method in BroadcastManagerBase are useful (although at this stage I'm happy to leave them there for now)

Reply:
>This condition shouldn't matter really as the same transaction can be processed as many times as we want but ProcessTransaction is idempotent so it won't change the state of the wallet.

Can you please elaborate? 
It looks like WalletManager.ProcessTransaction will call AddTransactionToWallet which will create new tx in the wallet OR update existing one. So if the only thing that changed is .IsPropagated flag then it's the only thing that will be updated.

Reply:
Exactly, what I'm saying is that ProcessTransaction can be called multiple times, and if there is a transaction that exists already in the wallet, it'll just update it.
so here, whether `if (transactionEntry.State == State.Propagated)` is true or false, we can call ProcessTransaction anyway.

Reply:
Makes sense, thanks!